apollo-federation/src/sources/connect/json_selection/parser.rs

```diff
@@ -191,28 +198,51 @@ impl NamedSelection {
         })
     }
 
+    // Parses either NamedPathSelection or PathWithSubSelection.
     fn parse_path(input: Span) -> IResult<Span, Self> {
-        tuple((Alias::parse, PathSelection::parse))(input)
-            .map(|(input, (alias, path))| (input, Self::Path(alias, path)))
+        let (remainder, (alias_opt, path)) =
+            tuple((opt(Alias::parse), PathSelection::parse))(input)?;
+
+        if alias_opt.is_none() && !path.has_subselection() {
+            Err(nom::Err::Error(nom::error::Error::new(
+                input,
+                nom::error::ErrorKind::IsNot,
+            )))
```

Comment:
This is where we enforce that a `PathSelection` used alongside other `NamedSelection` items has either an `Alias` or a `SubSelection`. This error leaves a lot to be desired, but I have a separate plan for improving parsing error messages (coming soon, assuming it works).

Reply:
do we have a ticket tracking your error plans?